Source/PluginProcessor.cpp

```diff
@@ -1513,14 +1517,21 @@ void PluginProcessor::receiveSysMessage(String const& selector, std::vector<pd::
                 }
             }
             
-            if (!editors.isEmpty()) {
+            if  (!editors.isEmpty()) {
                 auto* editor = editors[0];
                 if (auto* cnv = editor->getCurrentCanvas()) {
-                    editor->getTabComponent().openInPluginMode(cnv->patch);
+                    if(pluginModeFloatArgument)
+                        editor->getTabComponent().openInPluginMode(cnv->patch);
+                    else
+                        if (editor->isInPluginMode())
+                            editor->pluginMode->closePluginMode();
                 }
             } else {
-                patches[0]->openInPluginMode = true;
-            }
+                if(pluginModeFloatArgument)
+                    patches[0]->openInPluginMode = true;
+                else
+                    patches[0]->openInPluginMode = false;

```

Comment:
actually, I don't know if this line is necessary

Reply:
I mean I don't know when this code is executed, as in the `if  (!editors.isEmpty()) ` line before (l. 1520), the condition is true in my tests.

Reply:
You could send a `[pluginmode 0(` message.
Might make sense if a user does `[pluginmode $1(` and sends `0` or `1` in?